azure_img_utils/azure_image.py

```diff
@@ -480,6 +481,29 @@ def get_offer_doc(
         )
         return response
 
+    def submit_request(
+        self,
+        resource
+    ):
+        """
+        Submit a configuration request and wait for operation to finish
+
+        If the operation fails raise an exception.
+        """
+        headers = get_cloud_partner_api_headers(self.access_token)
+        job_id = submit_configure_request(headers, resource)
+
+        operation = self.wait_on_operation(job_id)
```

Comment:
Are we reasonably confident that this takes a few minutes or hours and not days?

Reply:
Yes, these are basically synchronous calls masquerading as async calls. The operations finish almost immediately. It's Azure so there's probably a chance for something to get stuck so maybe we want an exit clause (if it takes too long fail out).

Reply:
SO if we expect a couple of second completion than a 10 second initial sleep with a back of to 5 and then 2 seconds should give us enough time. That would be a total wait time of 17 seconds. Is that sufficient comfort to compensate for Azure unreliability?

Reply:
Given the expectation is for almost no wait maybe exponential backoff is better? And not sure what's too long to wait? In this case the much higher risk is not waiting long enough.

Reply:
Ultimately we just have to pick something. If an operation that is expected to take a few seconds ends up still waiting in 3 days then something is broken. So the question is when do we call it broken, 1 hour, 2 hours, a day? And yes we can make the wait period longer but per iteration but that would be counter intuitive, at least to me, since we already waited a longer period of time on an operation that we expect to complete quickly. Said differently, if I wait for an operation I expect to complete in 2 seconds for 10 seconds initially and then it is not completed when I come around again 5 seconds later I have waited a total of 15 seconds. If I assume the operation completed after 10 but in less than 15 seconds. If I revert that approach, of course it depends on the units we choose, I get the same effect if I initially wait 5 seconds and then 10 seconds on the second try. But that no longer holds true for the third iteration where we'd potentially wait 20 seconds in an increasing algorithm then I would have waited 35 seconds. Which is why I prefer decreasing algorithms in cases like this. But you have more experience with the actual operation time of the requests.

Reply:
We will need more data to be sure but in limited testing these operations are synchronous but coded asynchronously. E.g.

- We submit an action and get an operation ID back to check status
- Operation is performed almost immediately and thus the operation status check is "completed" within a couple seconds

Because we expect these operations to finish without delay exponential backoff seems the better option. That way we check frequently at the start and backoff if it's taking longer than expected. Without any data of this happening it's tough to gauge what would be a safe window to give up on the operation. Thus I would lean towards a rather large timeout initially as it's safer to wait longer than to fail out which leads to messy manual cleanup.

Reply:
Okay, added a timeout with default wait set to 10 minutes. It uses exponential backoff strategy since we expect the operation to finish almost immediately.